prometheusbp/promtest/testutil.go

```diff
@@ -1,13 +1,20 @@
 package promtest
 
 import (
+	"math"
 	"testing"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/testutil"
 	dto "github.com/prometheus/client_model/go"
 )
 
+const float64EqualityThreshold = 1e-9
+
+func almostEqual(a, b float64) bool {
+	return math.Abs(a-b) <= float64EqualityThreshold
+}
+
```

Comment:
This feels like it could still run into some issues if the floats are smaller. I gave this here a read https://floating-point-gui.de/errors/comparison/

I believe one way to deal with this is to use `big.Float` in Go. If this is okay though for the purpose of `CheckDelta` it's probably okay to just define it within `CheckDelta`?